app/services/legacy/dataset_index_service.rb

```diff
@@ -8,4 +8,17 @@ def index(legacy_dataset_id)
                                                id: dataset.id, body: dataset.as_indexed_json
                                              })
   end
+
+  def remove_from_index(legacy_dataset_id)
+    dataset = Dataset.find_by!(uuid: legacy_dataset_id)
+
+    begin
+      dataset.__elasticsearch__.delete_document({
+                                                index: ::Dataset.__elasticsearch__.index_name,
+                                                type: ::Dataset.__elasticsearch__.document_type,
+                                                id: dataset.id
+                                              })
+    rescue Elasticsearch::Transport::Transport::Errors::NotFound
```

Comment:
Does this mean we don't care about failed removals? Perhaps we could add a log just to say there was a failure.

Reply:
We generally don't (because it might not be there and we don't want to round-trip), but I'll add a .warn